GitCommands/Git/GitCreateTagCmd.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands.Git
+{
+    public enum TagOperation
+    {
+        Lightweight = 0,
+        Annotate,
+        SignWithDefaultKey,
+        SignWithSpecificKey
+    };
+
+    public static class TagOperationExtensions
+    {
+        public static bool CanProvideMessage(this TagOperation operationType)
+        {
+            switch (operationType)
+            {
+                case TagOperation.Lightweight:
+                    return false;
+                case TagOperation.Annotate:
+                case TagOperation.SignWithDefaultKey:
+                case TagOperation.SignWithSpecificKey:
+                    return true;
+                default:
+                    throw new NotSupportedException("Invalid TagOperation: " + operationType);
```

Comment:
How important is that? (and why?) It is kind of personal preference. Do we want to care about things like this?

Reply:
In a lot of cases string interpolation makes code more readable and maintainable.

However doing a bit a research on a subject (thank you) I was shocked to find out that interpolation is actually quite inefficient ([^1](http://www.tomdupont.net/2017/03/c-string-interpolation-performance.html), [^2](https://www.reddit.com/r/csharp/comments/6cpxjy/string_interpolation_vs_string_format_string/)). 
This said, that inefficiency will hardly have any profound effect on the app's performance in general, because most of the time the app is either waiting for user's input or git command to finish.

Reply:
>In a lot of cases string interpolation makes code more readable and maintainable.

It is very subjective, I would like to let devs decide which form they prefer to use. If a dev has a strong background in other language which does not have string interpolation then the dev may feel more confident using language structures he is familiar with. For instance it is not obvious that line such as  `_module.Received(1).RunGitCmd($"tag -f {switches} \"{TagName}\" -- \"{Revision}\"");` https://github.com/gitextensions/gitextensions/commit/9fb22f58727408d73baaea03ecf6cb69a66f997a#diff-3a91b3e9edc1faa5190d69a2fec75400R91
will produce double space between arguments for TagOperation.Lightweight

Reply:
Like I said - interpolation makes it simpler in some cases. In others it is an overhead.